### PR TITLE
Add login and registration, as well as sessions

### PR DESCRIPTION
**Pull request by thelastnode**
Users can be used through `req.user` and sessions via `req.session`.

### CODE REVIEW

File: lib/auth.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+                    first: attribs.userParams.first_name,
+                    last: attribs.userParams.last_name,
+                },
+                email: attribs.userParams.email,
+                is_admin: false,
+                gt_id: attribs.userParams.gt_id,
+                gt_userid: attribs.login,
+                password: hash,
+            });
+            user.save(function(err) {
```
Comment (ash211) [inline lib/auth.js:70]: Need to handle error condition here on failed saves

Comment (thelastnode) [inline lib/auth.js:70]: Good catch!

**Comment by ash211** [comment]
This commit adds a dependency on a conf.local.js file, which isn't in the git repo.  Can we at least make a template at conf.local.sample.js that can be filled in with true values?

So far I've seen the need for mongo.uri and session.secret config variables.

**Comment by thelastnode** [comment]
There's a pending issue (#5) for that.

**Pull request merged**